lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -730,28 +732,35 @@ def customize_vm(self, vm_obj):
         # Network settings
         adaptermaps = []
         for network in self.params['networks']:
-            if 'ip' in network and 'netmask' in network:
-                guest_map = vim.vm.customization.AdapterMapping()
-                guest_map.adapter = vim.vm.customization.IPSettings()
+            guest_map = vim.vm.customization.AdapterMapping()
+            guest_map.adapter = vim.vm.customization.IPSettings()
```

Comment:
you force the adapter creation if we don't set ip/netmask, this is wrong.

In my usecase we don't use this VMWare device